app/controllers/gift_requests_controller.rb

```diff
@@ -1,7 +1,12 @@
 class GiftRequestsController < ApplicationController
+  before_filter :authenticate_user!
```

Comment:
`before_filter` will be deprecated soon - `before_action` is equivalent and should be used instead